packages/kbn-i18n/src/react/inject_i18n_provider.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+
+import { I18nProvider } from './provider';
+
+export const injectI18nProvider = <P extends object>(WrappedComponent: React.ComponentType<P>) => {
+  const I18nProviderWrapper: React.SFC<P> = props => {
+    return (
+      <I18nProvider>
+        <WrappedComponent {...props} />
+      </I18nProvider>
+    );
+  };
+
+  I18nProviderWrapper.propTypes = WrappedComponent.propTypes;
```

Comment:
@azasypkin we need to pass to new wrapper component the same propTypes as wrapped component has because used `reactDirective` angular service from this library [ngReact](https://github.com/ngReact/ngReact) use propTypes of converted react component

> The reactDirective service will read the React component propTypes and watch attributes with these names. If your react component doesn't have propTypes defined you can pass in an array of attribute names to watch. If you don't pass any array of attribute names, fall back to use directive attributes as a last resort. By default, attributes will be watched by value however you can also choose to watch by reference or collection by supplying the watch-depth attribute. Possible values are reference, collection and value (default).

Reply:
__nit:__ please leave this comment in the code, right before we're reassigning `propTypes` including compacter version of excerpt from `reactDirective` docs (just first sentence probably). Smth like: we need to re-expose original `propTypes` from the wrapped component since it will be used by `reactDirective` Angular service that will rely on `propTypes` to watch attributes with these names.